pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -876,11 +881,16 @@ public ResultSet getResultSetImpl(long index, int count, Map<String, Class<?>> m
   public String toString() {
     if (fieldString == null && fieldBytes != null) {
       try {
-        Object array = readBinaryArray(1,0);
-        java.sql.Array tmpArray = connection.createArrayOf(getBaseTypeName(), (Object[]) array);
-        fieldString = tmpArray.toString();
+        Object array = readBinaryArray(1, 0);
+
+        if (PrimitiveArraySupport.isSupportedPrimitiveArray(array)) {
+          fieldString = PrimitiveArraySupport.arrayToString(connection.getTypeInfo().getArrayDelimiter(oid), array);
```

Comment:
might be like this:
```java
arraySupport = PrimitiveArraySupport.getArraySupport(array)
if (arraySupport != null) {
   fieldString = arraySupport.toArrayString(array)
}
```

The static `PrimitiveArraySupport.arrayToString()` method can be removed then.